### PR TITLE
Create categories

### PR DESCRIPTION
**Pull request by dougdroper**

### CODE REVIEW

File: db/migrate/20150129120103_add_default_to_category_ordinal.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddDefaultToCategoryOrdinal < ActiveRecord::Migration
+  def change
+    change_column :comfy_cms_categories, :ordinal, :integer, default: 999
```
Comment (benlovell) [inline db/migrate/20150129120103_add_default_to_category_ordinal.rb]: :cop: :oncoming_police_car:

File: app/views/categories/_form.html.haml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.l-panel-content
+  .l-constrained
+    = comfy_form_for @category, url: {action: "create"}, html: {class: 'form form--vertical'}, label_col: "form__label-heading", control_col: "form__input-wrapper" do |f|
+      = f.collection_select :parent_id, Comfy::Cms::Category.where(site_id: 1).reorder(:label), :id, :label, selected: @category.parent_id, include_blank: true
```
Comment (benlovell) [inline app/views/categories/_form.html.haml]: Did you need to specify `selected` here, it should just be inferred?

File: app/views/categories/_form.html.haml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.l-panel-content
+  .l-constrained
+    = comfy_form_for @category, url: {action: "create"}, html: {class: 'form form--vertical'}, label_col: "form__label-heading", control_col: "form__input-wrapper" do |f|
```
Comment (benlovell) [inline app/views/categories/_form.html.haml]: You got some double quoting action here.

File: app/views/categories/show.html.haml [hunk 4]
```diff
@@ -34,6 +34,9 @@
             .form-group
               .form__input-wrapper
                 = f.submit
+            .form-group
+              .form__input-wrapper
+                = button_to "Delete", category_path(@category), method: :delete, data: { confirm: "Are you sure?" }
```
Comment (benlovell) [inline app/views/categories/show.html.haml]: Double quotes here too.

File: app/models/comfy/cms/category.rb [hunk 5]
```diff
@@ -1,6 +1,9 @@
 require ComfortableMexicanSofa::Engine.root.join('app', 'models', 'comfy', 'cms', 'category')
 
 class Comfy::Cms::Category < ActiveRecord::Base
+  validates_presence_of :label, :title_en, :title_cy
```
Comment (benlovell) [inline app/models/comfy/cms/category.rb]: You could stack these up a la `validate :label, presence: true, uniqueness: true #...` did you chose not to? I don't feel strongly about it either way, just wondering :+1:

File: app/views/categories/_form.html.haml (hunk 3 above)
Comment (dougdroper) [inline app/views/categories/_form.html.haml]: I swear I am blind to these.. one day I won't have double quote fingers!

File: app/views/categories/_form.html.haml (hunk 2 above)
Comment (dougdroper) [inline app/views/categories/_form.html.haml]: actually pointless anyway, this is a new category (for the time being) :+1:

File: app/views/categories/_form.html.haml (hunk 3 above)
Comment (benlovell) [inline app/views/categories/_form.html.haml]: :smile:

File: app/models/comfy/cms/category.rb (hunk 5 above)
Comment (dougdroper) [inline app/models/comfy/cms/category.rb]: I don't have a strong bias but this way did read easier for me.

Comment (benlovell) [inline app/models/comfy/cms/category.rb]: Cool :+1:

File: db/migrate/20150129120103_add_default_to_category_ordinal.rb (hunk 1 above)
Comment (dougdroper) [inline db/migrate/20150129120103_add_default_to_category_ordinal.rb]: @benlovell Yeah this was a cheap way to ensure new records don't interfere with the order of anything without resorting to callbacks, and I was fairly confident that there wouldn't be more then 999 pages in any one category. I will have another think about a better way to do it though.

Comment (benlovell) [inline db/migrate/20150129120103_add_default_to_category_ordinal.rb]: Don't take me seriously, this was fine. My emoji were reference to the `999` being the number you dial for police here :smile: Leave it as-is!

Comment (dougdroper) [inline db/migrate/20150129120103_add_default_to_category_ordinal.rb]: Ahh, I read it as 'you should be thrown in jail' :) No worries will leave as is.

**Comment by tomas-stefano** [comment]
:+1:

**Comment by benlovell** [comment]
:shipit: :+1:

File: app/views/categories/index.html.haml [hunk 6]
```diff
@@ -4,6 +4,7 @@
       .l-panel-content__col
         %h1 Category Manager
       .l-panel-content__col
+        = link_to 'Create category', new_category_path
```
Comment (ourmaninamsterdam) [inline app/views/categories/index.html.haml]: Were we not going to put a button here?

File: app/views/categories/show.html.haml [hunk 7]
```diff
@@ -18,7 +18,7 @@
       .l-split-content
         .l-split-content__col
           %h2 Category details
-          = comfy_form_for @category, url: {action: "update"}, html: {class: 'form form--vertical'}, label_col: "form__label-heading", control_col: "form__input-wrapper" do |f|
+          = comfy_form_for @category, url: {action: 'update'}, html: {class: 'form form--vertical'}, label_col: 'form__label-heading', control_col: 'form__input-wrapper' do |f|
```
Comment (ourmaninamsterdam) [inline app/views/categories/show.html.haml]: We should think about moving this over to `form_for` at some point.

File: app/views/categories/index.html.haml (hunk 6 above)
Comment (dougdroper) [inline app/views/categories/index.html.haml]: will add it next iteration

File: app/views/categories/show.html.haml (hunk 7 above)
Comment (dougdroper) [inline app/views/categories/show.html.haml]: will do

File: app/views/categories/index.html.haml (hunk 6 above)
Comment (ourmaninamsterdam) [inline app/views/categories/index.html.haml]: :+1:

File: app/views/categories/_form.html.haml [hunk 8]
```diff
@@ -0,0 +1,19 @@
+.l-panel-content
+  .l-constrained
+    = comfy_form_for @category, url: {action: 'create'}, html: {class: 'form form--vertical'}, label_col: 'form__label-heading', control_col: 'form__input-wrapper' do |f|
```
Comment (ourmaninamsterdam) [inline app/views/categories/_form.html.haml]: Ditto `form_for`. It ain't broke, so up to you...

**Pull request merged**